Security/ObjectUserProvider.php

```
@@ -2,31 +2,41 @@
 
 namespace SumoCoders\FrameworkMultiUserBundle\Security;
 
+use SumoCoders\FrameworkMultiUserBundle\Exception\RepositoryNotRegisteredException;
+use SumoCoders\FrameworkMultiUserBundle\User\User;
+use SumoCoders\FrameworkMultiUserBundle\User\UserRepositoryCollection;
 use Symfony\Component\Security\Core\User\UserProviderInterface;
 use Symfony\Component\Security\Core\User\UserInterface;
 use Symfony\Component\Security\Core\Exception\UnsupportedUserException;
-use SumoCoders\FrameworkMultiUserBundle\User\UserRepository;
 use Symfony\Component\Security\Core\Exception\UsernameNotFoundException;
 
 class ObjectUserProvider implements UserProviderInterface
 {
-    /** @var UserRepository */
-    private $userRepository;
+    /** @var UserRepositoryCollection */
+    private $userRepositoryCollection;
 
     /**
-     * @param UserRepository $userRepository
+     * @param UserRepositoryCollection $userRepositoryCollection
      */
-    public function __construct(UserRepository $userRepository)
+    public function __construct(UserRepositoryCollection $userRepositoryCollection)
     {
-        $this->userRepository = $userRepository;
+        $this->userRepositoryCollection = $userRepositoryCollection;
     }
 
     /**
      * {@inheritDoc}
      */
     public function loadUserByUsername($username)
     {
-        $user = $this->userRepository->findByUsername($username);
+        $user = null;
```

isn't it better to throw an exception when no user is found